scripts/host-testbench.sh

```diff
@@ -56,64 +105,61 @@ rm -rf ./*.raw
 head -c ${INPUT_FILE_SIZE} < /dev/zero > zeros_in.raw
 
 # test with volume
-echo "=========================================================="
-echo "test volume with ./volume_run.sh 16 16 48000 zeros_in.raw volume_out.raw"
-if ./volume_run.sh 16 16 48000 zeros_in.raw volume_out.raw &>vol.log; then
-  echo "volume test passed!"
-else
-  echo "volume test failed!"
-  cat vol.log
-  exit 1
-fi
-
-if comparesize ${INPUT_FILE_SIZE} "$(filesize volume_out.raw)"; then
-  echo "volume_out size check passed!"
-else
-  echo "volume_out size check failed!"
-  cat vol.log
-  exit 1
-fi
-rm volume_out.raw vol.log
+ALGO=volume
+CMD="./${ALGO}_run.sh -v -b 16 zeros_in.raw ${ALGO}_out.raw"
+test_comp "$CMD" "$ALGO"
+
+ALGO=volume
+CMD="./${ALGO}_run.sh -v -b 24 zeros_in.raw ${ALGO}_out.raw"
+test_comp "$CMD" "$ALGO"
+
+ALGO=volume
+CMD="./${ALGO}_run.sh -v -b 32 zeros_in.raw ${ALGO}_out.raw"
+test_comp "$CMD" "$ALGO"
+
+# test with eqiir
+ALGO=eqiir
+CMD="./${ALGO}_run.sh -v -b 16 zeros_in.raw ${ALGO}_out.raw"
+test_comp "$CMD" "$ALGO"
+
+# test with eqfir
+ALGO=eqfir
+CMD="./${ALGO}_run.sh -v -b 24 zeros_in.raw ${ALGO}_out.raw"
+test_comp "$CMD" "$ALGO"
+
+# test with dcblock
+ALGO=dcblock
+CMD="./${ALGO}_run.sh -v -b 32 zeros_in.raw ${ALGO}_out.raw"
+test_comp "$CMD" "$ALGO"
+
+# test with DRC
+ALGO=drc
+CMD="./${ALGO}_run.sh -v -b 16 zeros_in.raw ${ALGO}_out.raw"
+test_comp "$CMD" "$ALGO"
+
+# test with Multi-band DRC
+ALGO=multiband-drc
+CMD="./${ALGO}_run.sh -v -b 32 zeros_in.raw ${ALGO}_out.raw"
+test_comp "$CMD" "$ALGO"
+
+# test with tdfb
+ALGO=tdfb
+CMD="./${ALGO}_run.sh -v -b 16 zeros_in.raw ${ALGO}_out.raw"
+test_comp "$CMD" "$ALGO"
```

Comment:
This should be a loop.

Reply:
This part of patch was removed, I'll do another PR to add more components to this test.